### ISSUE TITLE
Feature Request: Return elevation data in OSRM response

### ISSUE THREAD

**Issue by Redy1908** [Project-OSRM/osrm-backend#6846]
# Feature

Please describe the feature you would like to see in OSRM.
Images are often a good way to illustrate your requested feature.

When [integrating third party raster data](https://github.com/Project-OSRM/osrm-backend/wiki/Integrating-third-party-raster-data) it would be advantageous to incorporate elevation data for the respective locations in the response. This information could be leveraged in numerous ways, such as plotting a graph to illustrate the elevation trend along the route.

The following is a `STEP` extracted from an OSRM response with the added elevation field:

```json
{
   "geometry":"...",
   "maneuver":{
      "bearing_after":299,
      "bearing_before":0,
      "location":[
         "lat",
         "long",
         ELEVATION
      ],
      "modifier":"right",
      "type":"depart"
   },
   "mode":"driving",
   "driving_side":"right",
   "name":"Street Name",
   "intersections":[
      {
         "out":0,
         "entry":[
            true
         ],
         "bearings":[
            299
         ],
         "location":[
            "lat",
            "long",
            ELEVATION
         ]
      }
   ],
   "weight":71.4,
   "duration":22,
   "distance":141.3
}
```

**Comment by erlenddahl** [comment]
I need this as well, did you ever implement it, @Redy1908?

If not, I can probably hack it in place for my own use, but if any of the maintainers want me to make it a pull request, and have any pointers as to where and how it should be done, please let me know.

**Comment by Redy1908** [comment]
@erlenddahl Yes, but given the need for a fast solution, i decided not to hack the C++ layer. Instead i utilized PostGIS, an extension for PostgreSQL that allows for the manipulation of geospatial data. After importing the elevation data into PostGIS i queried the elevation data using the overview polyline provided by OSRM as input: 
```sql
SELECT ST_Value(rast, (dp).geom) As val
FROM (SELECT (ST_DumpPoints(ST_GeomFromText('OSRM_POLYLINE', 4326))) as dp) as points, dem
WHERE ST_Intersects(rast, (dp).geom)
ORDER BY (dp).path[1];
```
The query will return the list of the elevations of all the points inside the polyline

**Comment by erlenddahl** [comment]
Thank you for the speedy and detailed response! Perhaps that's easier in my case as well. Looks like we can make OSRM return the node ids as part of the response, which could perhaps be a faster way of retrieving the relevant lines from the database than using ST_Intersects.
